test/prom-query-tests/ExpressionQueriesIntegrationTests.cs

```diff
@@ -83,6 +103,124 @@ public class ExpressionQueriesIntegrationTests {
     Assert.Equal(sample3Value, sample3MetricValue); // Tests if our randomly generated sample 3 Value equals Prom's value
   }
 
+  private async Task QueryRangeTestHelper(Func<PrometheusClient, String, DateTime, DateTime, TimeSpan, Task<ResponseEnvelope<QueryResults>>> runQueryRange){
+    using var source = new CancellationTokenSource();
+    var token = source.Token;
+    var seed = DateTime.UtcNow;
+    var lastSampleTime = seed.Subtract(TimeSpan.FromMinutes(1));
+
+    // push test data
+    var samples = CreateTestData(lastSampleTime, 4);
+
+    // get data for sample out of range
+    var firstSample = samples.First();
+    var firstSampleTime = ConvertSampleTimestamp(firstSample.Timestamp);
+
+    // get data for second to last sample
+    var sample2 = samples[samples.Count() - 2];
+    var sample2RangeValue = sample2.Value;
+
+    // get data for last sample
+    var lastsample = samples.Last<Sample>();
+    var lastsampleRangeValue = lastsample.Value;
+    var lastsampleRangeTimestamp = ConvertSampleTimestamp(lastsample.Timestamp);
+
+    // create timespan
+    TimeSpan step = TimeSpan.FromMinutes(1);
+
+    // push test data, save random metric name
+    var metricName = await PushTestData(samples, token);
+
+    // initialize PrometheusClient
+    var queryClient = new PrometheusClient(CreateHttpClient);
+
+    // Q1: Out of range test EARLY
+    var queryEarlyOutOfRangeResult = await runQueryRange(
+      queryClient,
+      $"{metricName}",
+      firstSampleTime.AddMinutes(-4),
+      firstSampleTime.AddMinutes(-3),
+      step
+    );
+    Assert.NotNull(queryEarlyOutOfRangeResult.Data); // Tests if the query out of range result data is present
+    Assert.Empty(queryEarlyOutOfRangeResult.Data.Result); // Tests if the query out of range result is empty
+
+    // Q2: Out of range test LATE
```

Comment:
Could you add a comment here that describes the issue we had with the result still being returned even though the query was past the samples? I know you already put a note on 152 and 153 but I think a more detailed description at the top of this query would be helpful for people and you can remove the comments on 152 and 153.